django/db/models/sql/compiler.py

```diff
@@ -143,8 +143,14 @@ def collapse_group_by(self, expressions, having):
             # then also add having expressions to group by.
             pk = None
             for expr in expressions:
-                if (expr.output_field.primary_key and
-                        getattr(expr.output_field, 'model') == self.query.model):
+                # Is this a reference to query's base table primary key? If the expression
+                # isn't a Col-like, then skip the expression.
+                try:
+                    target = expr.target
+                    alias = expr.alias
+                except AttributeError:
+                    continue
```

Comment:
Not sure if this try-except is the most elegant way to pass non-col-like objects.